examples/tutorials/nb_python/Habitat_Interactive_Tasks.py

```diff
@@ -0,0 +1,2168 @@
+# 2. **Create a New Dataset**: We will define a new dataset class to save / load a list of episodes for the agent to train and evaluate on.
+# 3. **Grab / Release Action**: We will add the "grab/release" action to the agent's action space to allow the agent to pickup / drop an object under a crosshair.
+# 4. **Extend the Simulator Class**: We will extend the Simulator Class to add support for new actions implemented in previous step and add other additional utility functions
+# 5. **Create a New Task**: Create a new task definition, implement new *sensors* and *metrics*.
+# 6. **Train an RL agent**: We will define rewards for this task and utilize it to train an RL agent using the PPO algorithm.
+#
+# Let's get started!
+
+# %%
+# @title Installation { display-mode: "form" }
+# @markdown (double click to show code).
+
+# !curl -L https://raw.githubusercontent.com/facebookresearch/habitat-sim/master/examples/colab_utils/colab_install.sh | NIGHTLY=true bash -s
+# %cd /content
+
+# !gdown --id 1Pc-J6pZzXEd8RSeLM94t3iwO8q_RQ853
+# !unzip /content/coda.zip -d /content/habitat-sim/data/scene_datasets
+
+# %%
+# @title Path Setup and Imports { display-mode: "form" }
+# @markdown (double click to show code).
+
+import gzip
+import json
+import math
+import os
+import random
+import sys
+import time
+
+# %cd /content/habitat-api
+## [setup]
+import attr
+import cv2
+import git
+import magnum as mn
+import numpy as np
+
+try:
+    import ipywidgets as widgets
+    from IPython.display import display as ipydisplay
+
+    # For using jupyter/ipywidget IO components
+    from ipywidgets import fixed, interact, interact_manual, interactive
+
+    HAS_WIDGETS = True
+except ImportError:
+    HAS_WIDGETS = False
+
+from typing import Any, Dict, List, Optional, Type
+
+# %matplotlib inline
+from matplotlib import pyplot as plt
+from PIL import Image
+
+import habitat
+import habitat_sim
+from habitat.config import Config
+from habitat.core.registry import registry
+from habitat_sim.utils import common as ut
+from habitat_sim.utils import viz_utils as vut
+
+if "google.colab" in sys.modules:
+    os.environ["IMAGEIO_FFMPEG_EXE"] = "/usr/bin/ffmpeg"
+
+repo = git.Repo(".", search_parent_directories=True)
+dir_path = repo.working_tree_dir
+data_path = os.path.join(dir_path, "data")
+output_directory = "data/tutorials/output/"  # @param {type:"string"}
+output_path = os.path.join(dir_path, output_directory)
+
+if not os.path.exists(output_path):
+    os.makedirs(output_path)
+
+
+# %%
+# @title Util functions to visualize observations
+# @markdown - `make_video_cv2`: Renders a video from a list of observations
+# @markdown - `simulate`: Runs simulation for a given amount of time at 60Hz
+# @markdown - `simulate_and_make_vid` Runs simulation and creates video
+
+
+def make_video_cv2(
```

Comment:
convert to new utility function once it lands

Reply:
As an update, it should have landed.

Reply:
I have modified the make_video_cv2 for my usecase.